### PR TITLE
Interacting with K8s from a Docker Extension

### PR DESCRIPTION
**Pull request by felipecruz91**
### Proposed changes

This PR adds a doc that explains how to interact with Kubernetes from a Docker Extension.

See: https://deploy-preview-15896--docsdocker.netlify.app/desktop/extensions-sdk/dev/kubernetes/

### CODE REVIEW

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+---
+title: Interacting with Kubernetes from a Docker Extension
+description: How to connect to a Kubernetes cluster from a Docker Extension
+keywords: Docker, Extensions, sdk, Kubernetes
+---
+
+At the time of writing, the Extensions SDK do not provide any API methods to _directly_ interact with the Docker Desktop managed Kubernetes cluster or any other created using other tools such as KinD. However, this page provides a way for you to use other SDK APIs to interactive _indirectly_ with a Kubernetes cluster from your Docker extension.
```
Comment (benja-M-1) [inline desktop/extensions-sdk/dev/kubernetes.md:7]: I am not sure about this one:
```suggestion
At the time of writing, the Extensions SDK do not provide any API methods to _directly_ interact with the Docker Desktop managed Kubernetes cluster or any other created using other tools such as KinD. However, this page provides a way for you to use other SDK APIs to interact _indirectly_ with a Kubernetes cluster from your Docker extension.
```

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+`localStorage` is one of the mechanisms of a browser's web storage. It allows users to save data as key-value pairs in the browser for later use.
+`localStorage` does not clear data when the browser (the extension pane) closes. This makes it ideal for persisting data when navigating out of the extension to another parts of Docker Desktop.
+
+```typescript
+localStorage.setItem("kubeconfig", kubeConfig);
+```
+
+```typescript
+localStorage.getItem("kubeconfig");
```
Comment (benja-M-1) [inline desktop/extensions-sdk/dev/kubernetes.md:147]: The same way we have a warning about using localstorage [here](https://docs.docker.com/desktop/extensions-sdk/dev/oauth2-flow/#step-three-store-the-access-token), shouldn't we have one here as well?

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+await ddClient.docker.cli.exec("run", [
+  "--rm",
+  "-v",
+  "my-vol:/tmp",
+  "alpine",
+  "/bin/sh",
+  "-c",
+  `"touch /tmp/.kube/config && echo '${kubeConfig?.stdout}' > /tmp/.kube/config"`,
```
Comment (benja-M-1) [inline desktop/extensions-sdk/dev/kubernetes.md:133]: Just a thought, I always struggle to make this work . The content of stdout may be difficult to output in a file this way due to some unescaped characters. I haven't find a good way to make it work 100% of the time.

File: desktop/extensions-sdk/dev/kubernetes.md (hunk 2 above)
Comment (benja-M-1) [reply desktop/extensions-sdk/dev/kubernetes.md:147]:
```md
> Warning
>
> Storing the access token in a client’s localStorage is a security risk as that storage is currently shared amongst all extensions.
{: .warning}
```

File: desktop/extensions-sdk/dev/kubernetes.md (hunk 1 above)
Comment (aevesdocker) [inline desktop/extensions-sdk/dev/kubernetes.md:7]:
```suggestion
Currently, the Extensions SDK does not provide any API methods to directly interact with the Docker Desktop managed Kubernetes cluster or any other created using other tools such as KinD. However, this page provides a way for you to use other SDK APIs to interact indirectly with a Kubernetes cluster from your Docker extension.
```

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
+---
+title: Interacting with Kubernetes from a Docker Extension
+description: How to connect to a Kubernetes cluster from a Docker Extension
+keywords: Docker, Extensions, sdk, Kubernetes
+---
+
+At the time of writing, the Extensions SDK do not provide any API methods to _directly_ interact with the Docker Desktop managed Kubernetes cluster or any other created using other tools such as KinD. However, this page provides a way for you to use other SDK APIs to interactive _indirectly_ with a Kubernetes cluster from your Docker extension.
+
+Upvote [this issue](https://github.com/docker/extensions-sdk/issues/181) in our GitHub repository to show interest to request an API for interacting with Docker Desktop managed Kubernetes.
```
Comment (aevesdocker) [inline desktop/extensions-sdk/dev/kubernetes.md:9]:
```suggestion
To request an API that directly interacts with Docker Desktop-managed Kubernetes, you can upvote [this issue](https://github.com/docker/extensions-sdk/issues/181) in our GitHub repository.
```

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+## Prerequisites
+
+### Enable Kubernetes
+
+You can use the built-in Kubernetes in Docker Desktop to start a Kubernetes single-node cluster.
+A `kubeconfig` file is a file used to configure access to Kubernetes when used in conjunction with the `kubectl` commandline tool (or other clients).
```
Comment (aevesdocker) [inline desktop/extensions-sdk/dev/kubernetes.md:16]:
```suggestion
A `kubeconfig` file is used to configure access to Kubernetes when used in conjunction with the `kubectl` commandline tool, or other clients.
```

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+Docker Desktop conveniently provides the user with a local preconfigured `kubeconfig` file and `kubectl` command within the user’s home area. It is a convenient way to fast-tracking access for those looking to leverage Kubernetes from Docker Desktop.
+
+## Ship the `kubectl` as part of the extension
+
+If your extension needs to interact with Kubernetes clusters, it is recommended to include the `kubectl` commandline tool as part of your extension. By doing this, users who install your extension will get `kubectl` installed in their host (providing compatibility across Windows, Mac, and Linux), especially those that didn't get the `kubectl` commandline tool as part of enabling Docker Desktop.
```
Comment (aevesdocker) [inline desktop/extensions-sdk/dev/kubernetes.md:21]:
```suggestion
If your extension needs to interact with Kubernetes clusters, it is recommended that you include the `kubectl` command line tool as part of your extension. By doing this, users who install your extension get `kubectl` installed in their host.
```

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+To find out how to ship the `kubectl` commandline tool for multiple platforms as part of your Docker Extension image, click [here](https://docs.docker.com/desktop/extensions-sdk/extensions/multi-arch/#adding-multi-arch-binaries).
```
Comment (aevesdocker) [inline desktop/extensions-sdk/dev/kubernetes.md:23]:
```suggestion
To find out how to ship the `kubectl` command line tool for multiple platforms as part of your Docker Extension image, see [Build multi-arch extensions](https://docs.docker.com/desktop/extensions-sdk/extensions/multi-arch/#adding-multi-arch-binaries).
```

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+## Examples
+
+### Check the Kubernetes API server is reachable
+
+Once the `kubectl` commandline tool is added to the extension image in the `Dockerfile`, and defined in the `metadata.json`, the Extensions framework will deploy `kubectl` to the users' host when the extension is installed.
```
Comment (aevesdocker) [inline desktop/extensions-sdk/dev/kubernetes.md:29]:
```suggestion
Once the `kubectl` commandline tool is added to the extension image in the `Dockerfile`, and defined in the `metadata.json`, the Extensions framework deploys `kubectl` to the users' host when the extension is installed.
```

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  "-o",
+  'custom-columns=":metadata.name"',
+  "--context",
+  "docker-desktop",
+]);
+```
+
+## Persisting the kubeconfig file
+
+Below there are different ways to persist and read the kubeconfig file from the host filesystem. Users can add/edit/remove Kubernetes context to the `kubeconfig` file at anytime.
```
Comment (aevesdocker) [inline desktop/extensions-sdk/dev/kubernetes.md:70]:
```suggestion
Below there are different ways to persist and read the kubeconfig file from the host filesystem. Users can add, edit, or remove Kubernetes context to the `kubeconfig` file at any time.
```

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+> Warning
+>
+> The `kubeconfig` file is very sensitive and if found can allow an attacker administrative access to the Kubernetes Cluster.
```
Comment (aevesdocker) [inline desktop/extensions-sdk/dev/kubernetes.md:74]:
```suggestion
> The `kubeconfig` file is very sensitive and if found can give an attacker administrative access to the Kubernetes Cluster.
```

File: desktop/extensions-sdk/dev/kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+  } catch (err) {
+    console.log("error", JSON.stringify(err));
+  }
+};
+```
+
+### Docker volume
+
+Volumes are the preferred mechanism for persisting data generated by and used by Docker containers. You can make use of them to persit the `kubeconfig` file.
+By persisting the `kubeconfig` in a volume you won't need to read the `kubeconfig` file again when the extension pane closes. This makes it ideal for persisting data when navigating out of the extension to another parts of Docker Desktop.
```
Comment (aevesdocker) [inline desktop/extensions-sdk/dev/kubernetes.md:110]:
```suggestion
By persisting the `kubeconfig` in a volume you won't need to read the `kubeconfig` file again when the extension pane closes. This makes it ideal for persisting data when navigating out of the extension to other parts of Docker Desktop.
```

**Review by aevesdocker — COMMENTED** [review]
Small lil' fixes, nothing major, rest of the doc is looking great 👍

**Comment by felipecruz91** [comment]
Thank you @aevesdocker for taking the time to review the full doc! 😄

File: desktop/extensions-sdk/dev/kubernetes.md (hunk 2 above)
Comment (felipecruz91) [reply desktop/extensions-sdk/dev/kubernetes.md]: At the moment we do not have a way to securely persist sensitive data, that's why the warning is at the beginning of the subsection: https://github.com/docker/docs/pull/15896/files#diff-93f5c7aa759604f80d718196fcedffed6913a42d1ba39e1e7c11edbd09bbfd6aR72-R75

**Review by aevesdocker — APPROVED** [review]
Thank you @felipecruz91!

**Pull request merged**